index.js

```diff
@@ -49,6 +49,55 @@ let allowLiveReload = function(policyObject, liveReloadConfig) {
 module.exports = {
   name: require('./package').name,
 
+  // Configuration is only available by public API in `app` passed to some hook.
+  // We calculate configuration in `config` hook and use it in `serverMiddleware`
+  // and `contentFor` hooks, which are executed later. This prevents us from needing to
+  // calculate the config more than once. We can't do this in `contentFor` hook cause
+  // that one is executed after `serverMiddleware` and can't do it in `serverMiddleware`
+  // hook cause that one is only executed on `ember serve` but not on `ember build` or
+  // `ember test`. We can't do it in `init` hook cause app is not available by then.
+  //
+  // The same applies to policy string generation. It's also calculated in `config`
+  // hook and reused in both others. But this one might be overriden in `serverMiddleware`
+  // hook to support live reload. This is safe because `serverMiddleware` hook is executed
+  // before `contentFor` hook.
+  //
+  // Only a small subset of the configuration is required at run time in order to support
+  // FastBoot. This one is returned here as default configuration in order to make it
+  // available at run time.
+  config: function(environment, runConfig) {
+    // calculate configuration and policy string
+    // hook may be called more than once, but we only need to calculate once
+    if (!this._config) {
+      let { app, project } = this;
+      let ownConfig = readConfig(project, environment);
+      let ui = project.ui;
+      let config = calculateConfig(environment, ownConfig, runConfig, ui);
+
+      // add static test nonce if build includes tests
+      // Note: app is not defined for CLI commands
+      if (app && app.tests) {
+        appendSourceList(config.policy, 'script-src', `'nonce-${STATIC_TEST_NONCE}'`);
+      }
+
+      this._config = config;
+      this._policyString = buildPolicyString(config.policy);
+    }
+
+    // provide configuration needed at run-time for FastBoot support (if needed)
+    // TODO: only inject if application uses FastBoot
+    if (!this._config.enabled || !this._config.delivery.includes('header')) {
```

Comment:
How does the delivery mode get set to header? Should have a specific opt-in for the FastBoot behavior (via either a special delivery mode, or something)?

Reply:
Also, should we make a constant for `DELIVERY_HEADER` (or change the usage of `DELIVERY_META` to just be `'meta'`, which seems better to me personally)? It's a tad odd that we treat them differently in different places...

Reply:
> How does the delivery mode get set to header?

This is the `delivery` configuration option. It's set via `config/content-security-policy.js`. It defaults to `["header"]`.

> Should have a specific opt-in for the FastBoot behavior (via either a special delivery mode, or something)?

I'm not sure. That's the open question in PR text. I tend to believe that `enabled` and `delivery` method gives enough control to disable setting for FastBoot. One might either remove `"header"` from `delivery` or disable the addon at all in production environment. I can't image any use case in which you need `delivery` to include `"header"` but don't want the header to be set in FastBoot. Beside FastBoot support the delivery method `"header"` only controls if CSP headers are set by Ember CLI development server. If I got @sandstrom right, he has the same feeling (https://github.com/rwjblue/ember-cli-content-security-policy/pull/113#issuecomment-536521562).

Reply:
> Also, should we make a constant for `DELIVERY_HEADER` (or change the usage of `DELIVERY_META` to just be `'meta'`, which seems better to me personally)? It's a tad odd that we treat them differently in different places...

Removed `DELIVERY_META` and `DELIVERY_HEADER` consts in b118a5dfc19b2baea194f8c7df389b192ac5b5a5.

Reply:
I would not mix that both concerns. There is also a delivery option `"meta"`, which causes the policy to be injected as a meta tag. This is totally valid concern regardless of FastBoot.

If the addon is enabled in production, consuming app does not use FastBoot and delivery does *not* include `"meta"`, the addon should not have any effect. If you are concerned about unnecessary run-time config, I would rather fix #116 before releasing a new version than having complex default due to it. Fixing #116 should not be more difficult than having defaults depending on the presence of ember-cli-fastboot.

Reply:
> If you are concerned about unnecessary run-time config

Yes, that is exactly my concern. My comment above should likely have _gone on #116_. 😸